### ISSUE TITLE
Flux OCIRepo Support in Zarf

### ISSUE THREAD

**Issue by cmwylie19** [defenseunicorns/zarf#2021]
### Is your feature request related to a problem? Please describe.
Zarf to be able to create packages that include Flux's [OCIRepo](https://fluxcd.io/flux/components/source/ocirepositories/) 

```yaml
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: OCIRepository
metadata:
  name: podinfo
  namespace: default
spec:
  interval: 5m0s
  url: oci://ghcr.io/stefanprodan/manifests/podinfo
  ref:
    tag: latest
 ```
### Describe the solution you'd like

- **Given** an OCIRepo is listed as a manifest
- **When** the zarf package with an OCIRepo is deployed
- **Then** the OCI manifests/images are transferred to Zarf's internal image registry for the airgap

### Additional context
Relates to #1974

**Comment by Noxsios** [comment]
Finishing this will require takeover of https://github.com/defenseunicorns/zarf/pull/2005

**Comment by koolay** [comment]
Hi, any plan ?

**Comment by AustinAbro321** [comment]
Implemented by #2005
